### PR TITLE
Unify working of guide quest with Shortcut

### PR DESCRIPTION
**Pull request by sonohoshi**
### Description

1. Now, guide quest doesn't pass to stage.
2. It works the same as Acquisition place button(Shortcut).

~~- **There may be variations in the log.**~~
~~`Unity/Click Guided Quest Enter Dungeon`~~
~~`Unity/Click Guided Quest Enter Event Dungeon`~~
~~Previously, there was Analyzer invoking at the same time as action tx.~~
~~Now, After analyzer invoking, player may not send a tx. We need check it.~~

### How to test

1. Click any stage guide quest or event dungeon quest.

### Related Links

[가이드 퀘스트 및 숏컷 동작 개선 및 리팩토링](https://app.asana.com/0/958521740385861/1202847454522981/f)

### Required Reviewers

@planetarium/9c-dev @jaeho0103 

### Screenshot
Honeycam 2022-08-25 17-37-51

### CODE REVIEW

File: nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs [hunk 1]
```diff
@@ -10,13 +10,24 @@
...
 
+        public enum PlaceType
+        {
+            Stage,
+            Shop = 3,
+            Arena = 4,
+            Quest = 5,
+            Staking = 7,
+            EventDungeonStage = 8,
+        }
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs:29]: Where are the numbers 1 and 2?

File: nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs [hunk 2]
```diff
@@ -245,13 +248,92 @@ or ItemSubType.MonsterPart
...
+        public static bool CheckConditionOfShortcut(PlaceType type, int stageId = 0)
+        {
+            switch (type)
+            {
+                case PlaceType.EventDungeonStage:
+                    if (RxProps.EventDungeonInfo.Value is not null)
+                    {
+                        return stageId <=
+                               RxProps.EventDungeonInfo.Value.ClearedStageId + 1;
+                    }
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs:270]: How about this? If you want to compare the `stageId` with a playable stage id.
Sometimes the `RxProps.EventDungeonInfo.Value.ClearedStageId` can be zero. And we should consider the end of stage.

```suggestion
                    var playableStageId = var openedStageId =
                        RxProps.EventDungeonInfo.Value is null ||
                        RxProps.EventDungeonInfo.Value.ClearedStageId == 0
                            ? RxProps.EventDungeonRow.StageBegin
                            : math.min(
                                RxProps.EventDungeonInfo.Value.ClearedStageId + 1,
                                RxProps.EventDungeonRow.StageEnd);
                    return stageId <= playableStageId;
```

File: nekoyume/Assets/_Scripts/Analyzer.cs [hunk 3]
```diff
@@ -12,6 +12,8 @@ public class Analyzer
 
         private readonly MixpanelValueFactory _mixpanelValueFactory;
 
+        public int GuideQuestStageId { get; set; }
+
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/Analyzer.cs:16]: This variable seems not fit with the `Analyzer`. Can we pair about this?

File: nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs (hunk 1 above)
Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs:29]: Previously too, 1 and 2 were not used, but I think we need further explanation. I'll add comments.

File: nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs (hunk 2 above)
Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/Helper/ShortcutHelper.cs:270]: Sounds good! I didn't think about end of stage. I'll re-write code with considering that.

File: nekoyume/Assets/_Scripts/Analyzer.cs (hunk 3 above)
Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/Analyzer.cs:16]: OK I'll solve another problem and give you a mention. thx

**Review by boscohyun — APPROVED** [review]

**Pull request merged**